packages/@aws-cdk/aws-batch/lib/managed-compute-environment.ts

```diff
@@ -0,0 +1,1222 @@
+   * within a single Availability Zone with high network flow potential.
+   *
+   * @see: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html
+   *
+   * @default - no placement group
+   */
+  readonly placementGroup?: ec2.IPlacementGroup;
+}
+
+/**
+ * A ManagedComputeEnvironment that uses ECS orchestration on EC2 instances.
+ *
+ * @resource AWS::Batch::ComputeEnvironment
+ */
+export class ManagedEc2EcsComputeEnvironment extends ManagedComputeEnvironmentBase implements IManagedEc2EcsComputeEnvironment {
+  /**
+   * refer to an existing ComputeEnvironment by its arn.
+   */
+  public static fromManagedEc2EcsComputeEnvironmentArn(
+    scope: Construct, id: string, managedEc2EcsComputeEnvironmentArn: string,
+  ): IManagedEc2EcsComputeEnvironment {
+    const stack = Stack.of(scope);
+    const computeEnvironmentName = stack.splitArn(managedEc2EcsComputeEnvironmentArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
+
+    class Import extends Resource implements IManagedEc2EcsComputeEnvironment {
+      public readonly computeEnvironmentArn = managedEc2EcsComputeEnvironmentArn;
+      public readonly computeEnvironmentName = computeEnvironmentName;
+      public readonly enabled = true;
+      public readonly instanceClasses = [];
+      public readonly instanceTypes = [];
+      public readonly maxvCpus = 1;
+      public readonly connections = { } as any;
+      public readonly securityGroups = [];
+
+      public addInstanceClass(_instanceClass: ec2.InstanceClass): void {
+        throw new Error(`cannot add instance class to imported ComputeEnvironment '${id}'`);
+      }
+      public addInstanceType(_instanceType: ec2.InstanceType): void {
+        throw new Error(`cannot add instance type to imported ComputeEnvironment '${id}'`);
+      }
+    }
+
+    return new Import(scope, id);
+  }
+  public readonly computeEnvironmentArn: string;
+  public readonly computeEnvironmentName: string;
+
+  public readonly images?: EcsMachineImage[];
+  public readonly allocationStrategy?: AllocationStrategy;
+  public readonly spotBidPercentage?: number;
+  public readonly spotFleetRole?: iam.IRole;
+  public readonly instanceTypes: ec2.InstanceType[];
+  public readonly instanceClasses: ec2.InstanceClass[];
+  public readonly instanceRole?: iam.IRole;
+  public readonly launchTemplate?: ec2.ILaunchTemplate;
+  public readonly minvCpus?: number;
+  public readonly placementGroup?: ec2.IPlacementGroup;
+
+  private readonly instanceProfile: iam.CfnInstanceProfile;
+
+  constructor(scope: Construct, id: string, props: ManagedEc2EcsComputeEnvironmentProps) {
+    super(scope, id, props);
+
+    this.images = props.images;
+    this.allocationStrategy = determineAllocationStrategy(id, props.allocationStrategy, this.spot);
+    this.spotBidPercentage = props.spotBidPercentage;
+    this.spotFleetRole = props.spotFleetRole ?? (this.spot ? createSpotFleetRole(this) : undefined);
```

Comment:
This is _only_ required if `spot` and `allocationStrategy` is set explicitly to BEST_FIT or null (to api).  The spot SCO and BFP do not require it.